core_java/twitter/src/main/java/ca/jrvs/apps/twitter/controller/TwitterController.java

```diff
@@ -0,0 +1,91 @@
+package ca.jrvs.apps.twitter.controller;
+
+import ca.jrvs.apps.twitter.model.GeoLoc;
+import ca.jrvs.apps.twitter.model.Tweet;
+import ca.jrvs.apps.twitter.service.TwitterService;
+import java.util.Arrays;
+import java.util.List;
+
+public class TwitterController implements Controller {
+
+  private TwitterService twitterService;
+
+  // Default constructor
+  public TwitterController() {
+    twitterService = new TwitterService();
+  }
+
+  // Alternate constructor to provide your own TwitterService
+  public TwitterController(TwitterService twitterService) {
+    this.twitterService = twitterService;
+  }
+
+  /**
+   * Parse user argument and post a tweet by calling service classes.
+   *
+   * @param args Post contents and optionally a longitude and latitude
+   * @return a posted tweet
+   * @throws IllegalArgumentException if args are invalid
+   */
+  @Override
+  public Tweet postTweet(String[] args) {
+    Tweet tweetToPost = new Tweet();
+    if (args.length < 1) {
+      throw new IllegalArgumentException("At least 1 argument is required");
+    }
+    tweetToPost.setText(args[0]);
+    if (args.length == 3) {
+      String lon = args[1].replace(',', '.');
+      String lat = args[2].replace(',', '.');
+      try {
+        GeoLoc location = new GeoLoc();
+        location
+            .setCoordinates(new float[]{Float.parseFloat(lon), Float.parseFloat(lat)});
+        tweetToPost.setLocation(location);
+      } catch (NumberFormatException nfex) {
+        throw new IllegalArgumentException(
+            "Second and third arguments must be Longitude and Latitude");
+      }
+    } else if (args.length == 2 || args.length > 3) {
+      throw new IllegalArgumentException("Incorrect number of args given. Expected 1 or 3");
+    }
+    return twitterService.postTweet(tweetToPost);
+  }
+
+  /**
+   * Parse user argument and search a tweet by calling service classes.
+   *
+   * @param args A Tweet ID and optionally fields to include in the response
+   * @return a tweet
+   * @throws IllegalArgumentException if args are invalid
+   */
+  @Override
+  public Tweet showTweet(String[] args) {
+    String tweetId;
+    String[] fields = {};
+    if (args.length >= 1) {
+      tweetId = args[0];
+      if (args.length >= 2) {
```

Comment:
Seems inefficient to check the length twice.  We know from TwitterCLIApp that there are at least two arguments passed.  
I'd suggest just checking if the fields are present since we should already know that tweetId is there.